TestModels/ExternV2/src/ExternV2Constructor.dfy

```diff
@@ -0,0 +1,26 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+include "../Model/SimpleDafnyExternV2Types.dfy"
+
+replaceable module ExternV2Constructor {
+    import opened Wrappers
+    import opened SimpleDafnyExternV2Types
+
+    class{:extern "ExternV2ConstructorClass"} ExternV2ConstructorClass {
+
+        function {:extern} Value(): string reads this
+
+        // A build method is used instead of a constructor
+        // because in some native runtimes, constructor can throw.
+        // But Dafny constructors MUST succeed so this mismatch
+        // make a static Build method required for extern classes.
+        static method {:extern} Build(input: string)
+        returns (output: Result<ExternV2ConstructorClass, Error>)
+        ensures output.Success?
+        ==>
+            && output.value.Value() == input
+            && fresh(output.value)
+
+        method{:extern} GetValue() returns (output: Result<string, Error>)
+    }
```

Comment:
Question: Will the `replaces` modules change the name of the `ExternV2ConstructorClass`?

Reply:
No. This stays the same in generated code.

NET:

```
      _out0 = simple.dafnyexternv2.internaldafny.ExternV2Constructor.ExternV2ConstructorClass.Build(Wrappers_Compile.Option<Dafny.ISequence<char>>.UnwrapOr((input).dtor_value, Dafny.Sequence<char>.FromString("Error")));
```

Java:

```
    if ((_1_valueOrError0).IsFailure(((dafny.TypeDescriptor<simple.dafnyexternv2.internaldafny.ExternV2Constructor.ExternV2ConstructorClass>)(java.lang.Object)dafny.TypeDescriptor.reference(simple.dafnyexternv2.internaldafny.ExternV2Constructor.ExternV2ConstructorClass.class)), simple.dafnyexternv2.internaldafny.types.Error._typeDescriptor())) {

```

It looks like the only name that is replaced is on the replaceable module.